src/lib/core/components/visualizations/implementations/MapVisualization.tsx

```diff
@@ -0,0 +1,242 @@
+import * as t from 'io-ts';
+
+// map component related imports
+import MapVEuMap from '@veupathdb/components/lib/map/MapVEuMap';
+import { defaultAnimationDuration } from '@veupathdb/components/lib/map/config/map.json';
+import geohashAnimation from '@veupathdb/components/lib/map/animation_functions/geohash';
+import { BoundsViewport } from '@veupathdb/components/lib/map/Types';
+import DonutMarker from '@veupathdb/components/lib/map/DonutMarker';
+import { BoundsDriftMarkerProps } from '@veupathdb/components/lib/map/BoundsDriftMarker';
+
+// viz-related imports
+import { PlotLayout } from '../../layouts/PlotLayout';
+import { useDataClient, useStudyMetadata } from '../../../hooks/workspace';
+import { useMemo, useCallback, useState, ReactElement } from 'react';
+import { preorder } from '@veupathdb/wdk-client/lib/Utils/TreeUtils';
+import DataClient, { MapMarkersRequestParams } from '../../../api/DataClient';
+import { useVizConfig } from '../../../hooks/visualizations';
+import { usePromise } from '../../../hooks/promise';
+import {
+  filtersFromBoundingBox,
+  geohashLevelToVariableId,
+  leafletZoomLevelToGeohashLevel,
+} from '../../../utils/visualization';
+
+export const mapVisualization: VisualizationType = {
+  selectorComponent: SelectorComponent,
+  fullscreenComponent: MapViz,
+  createDefaultConfig: createDefaultConfig,
+};
+
+function SelectorComponent() {
+  return (
+    <img
+      alt="Geographic map"
+      style={{ height: '100%', width: '100%' }}
+      src={map}
+    />
+  );
+}
+
+function createDefaultConfig(): MapConfig {
+  return {
+    mapCenterAndZoom: {
+      latitude: 0,
+      longitude: 0,
+      zoomLevel: 1, // TO DO: check MapVEuMap minZoom hardcoded to 2
+    },
+  };
+}
+
+const defaultAnimation = {
+  method: 'geohash',
+  animationFunction: geohashAnimation,
+  duration: defaultAnimationDuration,
+};
+
+type MapConfig = t.TypeOf<typeof MapConfig>;
+// eslint-disable-next-line @typescript-eslint/no-redeclare
+const MapConfig = t.type({
+  mapCenterAndZoom: t.type({
+    latitude: t.number,
+    longitude: t.number,
+    zoomLevel: t.number,
+  }),
+});
+
+function MapViz(props: VisualizationProps) {
+  const {
+    computation,
+    visualization,
+    updateConfiguration,
+    updateThumbnail,
+    filters,
+    dataElementConstraints,
+    dataElementDependencyOrder,
+    starredVariables,
+    toggleStarredVariable,
+    totalCounts,
+    filteredCounts,
+  } = props;
+  const studyMetadata = useStudyMetadata();
+  const { id: studyId } = studyMetadata;
+  const entities = useMemo(
+    () =>
+      Array.from(preorder(studyMetadata.rootEntity, (e) => e.children || [])),
+    [studyMetadata]
+  );
+  const dataClient: DataClient = useDataClient();
+
+  const [vizConfig, updateVizConfig] = useVizConfig(
+    visualization.descriptor.configuration,
+    MapConfig,
+    createDefaultConfig,
+    updateConfiguration
+  );
+
+  const handleViewportChanged = useCallback(
+    ({ center, zoom }) => {
```

Comment:
```suggestion
    ({ center, zoom }: { center?: number[] | null; zoom?: number | null }) => {
```

I suggest adding typing to prevent `center` and `zoom` being `any`, though the inline type isn't very pretty

Reply:
That's weird - my emacs IDE doesn't highlight this as a problem/warning, though the type of the callback is
```
const handleViewportChanged: ({ center, zoom }: any) => void
```
So it should warn about `any`s.

I'll try and make a prettier fix, but don't want to import `react-leaflet`'s `Viewport` type into the client.  I think I have an idea.

Reply:
I made a [fix here](https://github.com/VEuPathDB/web-eda/pull/782/commits/a71986cf0a37208818dc23790ef9ad4b781026f2) but I don't really see why we have to do it. I'm sure I haven't had to do that kind of thing before...

Reply:
Ah, nice one! Yeah, IDK why eslint doesn't complain about those `any`s